core/blockly.js

```diff
@@ -173,9 +173,12 @@ Blockly.svgResize = function(workspace) {
  * @private
  */
 Blockly.onKeyDown_ = function(e) {
-  if (Blockly.mainWorkspace.options.readOnly || Blockly.utils.isTargetInput(e)) {
+  if (Blockly.mainWorkspace.options.readOnly || Blockly.utils.isTargetInput(e)
+      || (Blockly.mainWorkspace.rendered && !Blockly.mainWorkspace.isVisible)) {
```

Comment:
This looks like it could break in situations with multiple workspaces. What if `mainWorkspace` is the read-only / headless workspace, while other workspaces are able to accept input. Consider the multiple workspaces in the developer tools; the assignment of `mainWorkspace` is somewhat arbitrary.

At minimum, add a note in the function documentation or add a TODO to address this in the future.